mlflow/utils/conda.py

```diff
@@ -61,11 +61,22 @@ def get_conda_bin_executable(executable_name):
     return executable_name
 
 
-def _get_conda_env_name(conda_env_path, env_id=None):
-    conda_env_contents = open(conda_env_path).read() if conda_env_path else ""
+def _get_conda_env_name(conda_env_path, env_id=None, env_root_dir=None):
+    if conda_env_path:
+        with open(conda_env_path) as f:
+            conda_env_contents = f.read()
+    else:
+        conda_env_contents = ""
+
     if env_id:
         conda_env_contents += env_id
-    return "mlflow-%s" % hashlib.sha1(conda_env_contents.encode("utf-8")).hexdigest()
+
+    env_name = "mlflow-%s" % hashlib.sha1(conda_env_contents.encode("utf-8")).hexdigest()
+    if env_root_dir:
+        env_root_dir = os.path.normpath(env_root_dir)
+        env_name += "-%s" % hashlib.sha1(env_root_dir.encode("utf-8")).hexdigest()
```

Comment:
```suggestion
    hash_input = conda_env_contents + (os.path.normpath(env_root_dir) if env_root_dir else "")
    env_name = "mlflow-%s" % hashlib.sha1(hash_input.encode("utf-8")).hexdigest()
```

Can we create the env suffix like above?

Reply:
Or are we intentionally splitting the suffixes?

Reply:
I prefer to hash the "conda_env_contents" part and "env_root_dir" part separately, it is easier for debug, i.e,
"mlflow-{conda_env_contents_hash}-{env_root_dir_hash}"
If "conda_env_contents_hash" part is the same, it means it is conda env based on the same conda.yaml file.
This is useful for debugging.
But if we connect them and hash together, then according to the env name we cannot distinguish whether they are based on the same  conda.yaml file or not

Reply:
ok, can you add a comment to indicate that?